pkg/rules/engine/engine.go

```diff
@@ -307,3 +307,57 @@ func (engine *Engine) GetSelectedEvents() []detect.SignatureEventSelector {
 	}
 	return res
 }
+
+// GetFilters returns a mediated array of all signature's filters
+func (engine *Engine) GetFilters() ([]protocol.Filter, error) {
+	allFilters := make([]protocol.Filter, 0)
+	engine.signaturesMutex.RLock()
+	for sig := range engine.signatures {
+		filters, err := sig.GetFilters()
```

Comment:
why we return if only one signature fails in giving back its filters? why not just pushing error to error channel and continue with the iteration over the signatures?

Reply:
I remember adding a comment about that, looks like it went missing, i'll add it back too.

But basically, if even one signature has no filters, if we go ahead with filtering anyway, that signature will most likely miss a lot, if not all, relevant events, so filtering opts out the moment not all signatures have it enabled.